gnomad_qc/v4/create_release/calculate_variant_statistics.py

```diff
@@ -0,0 +1,535 @@
+    :param agg_rare_variants: Stratify by variants which have adj AF <0.1%.
+    :param suffix: String of arguments to append to name.
+    """
+
+    # Read in release data and metadata and VDS.
+    ht_release_data = release_sites(data_type).ht()
+    if test:
+        logger.info("Test: filtering to only chr22 sites and 467 test samples in VDS")
+        ht_release_data = ht_release_data.filter(
+            ht_release_data.locus.contig == "chr22"
+        )
+
+    if filter_variant_qc:
+        logger.info("Filter to those which pass all variant qc filters")
+        ht_release_data = ht_release_data.filter(hl.len(ht_release_data.filters) == 0)
+
+    meta_ht = meta(data_type=data_type).ht()
+    vds = get_gnomad_v4_vds(test=test, release_only=True, split=True)
+
+    # Filter to only variants and samples in release data.
+    logger.info("Filter to only variants and samples in gnomAD release.")
+    vds = hl.vds.filter_variants(
+        vds,
+        vds.variant_data.filter_rows(
+            ~hl.is_missing(ht_release_data[vds.variant_data.row_key])
+        )
+        .rows()
+        .select(),
+    )
+
+    vds = hl.vds.filter_samples(
+        vds,
+        vds.reference_data.filter_cols(meta_ht[vds.reference_data.s].release)
+        .cols()
+        .select(),
+    )
+
+    # Filter to variants only in requested calling intervals.
+    if filter_ukb_regions:
+        logger.info("Filter to only variants in UKB capture regions.")
+        filtered_ukb = (
+            vds.variant_data.filter_rows(
+                ht_release_data[
+                    vds.variant_data.row_key
+                ].region_flags.outside_ukb_capture_region
+            )
+            .rows()
+            .select()
+        )
+        vds = hl.vds.filter_variants(vds, filtered_ukb)
+
+    if filter_broad_regions:
+        logger.info("Filter to only variants in Broad capture regions.")
+        filtered_broad = (
+            vds.variant_data.filter_rows(
+                ht_release_data[
+                    vds.variant_data.row_key
+                ].region_flags.outside_broad_capture_region
+            )
+            .rows()
+            .select()
+        )
+        vds = hl.vds.filter_variants(vds, filtered_broad)
+
+    # Create variant matrix table from VDS and annotate with all relevant info
+    vmt = vds.variant_data
+    logger.info("VDS created and checkpointing, without annotations")
+    vmt = vmt.checkpoint(
+        new_temp_file(f'vmt_{"TEST" if test else ""}_mt', extension="mt"),
+        overwrite=overwrite,
+    )
+
+    # Add extra Allele Count and A-Type, according to Hail standards, to help their computation
+    ac_and_atype = vmt_sample_qc_variant_annotations(
+        global_gt=vmt.GT, alleles=vmt.alleles
+    )
+    vmt = vmt.annotate_rows(**ac_and_atype)
+
+    # NOTE: not size efficient, but we are never checkpointing THIS I don't believe
+    vmt = vmt.annotate_rows(**ht_release_data[vmt.row_key])
+
+    if agg_lof:
```

Comment:
This made me think: do you need to get stats for missense, synonymous variants or any specified CSQs?